Sources/Testing/Parameterization/Test.ID.CaseSelection.swift

```diff
@@ -0,0 +1,67 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension Test.ID {
+  /// A type representing a selection of test cases grouped by their associated
+  /// test.
+  struct CaseSelection: Sendable {
```

Comment:
I'm a bit confused why we need this _and_ `Test.Case.ID.Selection`. Can we simplify things?